### PR TITLE
Integrate porch with renderer interface

### PR DESCRIPTION
**Pull request by phanimarupaka**
This PR integrates porch renderer with kpt renderer.

### CODE REVIEW

File: pkg/fn/eval.go [hunk 1]
```diff
@@ -17,13 +17,19 @@ package fn
...
+	"sigs.k8s.io/kustomize/kyaml/kio"
 )
 
 type EvalOptions struct {
+	// FnResultList stores the result of the function evaluation
+	FnResultList fnresult.ResultList
 }
 
 type Evaluator interface {
 	Eval(ctx context.Context, pkg filesys.FileSystem, fn v1.Function, opts EvalOptions) error
```
Comment (phanimarupaka) [inline pkg/fn/eval.go:33]: @martinmaly Do we need this ? Can't we just leverage NewRunner function and kio.Filter interface ? Refer to fnruntime package.

Comment (martinmaly) [reply pkg/fn/eval.go:33]: it may work. What does pkgPath do? If we use kio.Filter there should be no need for paths.

File: pkg/fn/eval.go [hunk 2]
```diff
@@ -17,13 +17,19 @@ package fn
...
+	"github.com/GoogleContainerTools/kpt/internal/types"
+	fnresult "github.com/GoogleContainerTools/kpt/pkg/api/fnresult/v1"
 	v1 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1"
 	"sigs.k8s.io/kustomize/kyaml/filesys"
+	"sigs.k8s.io/kustomize/kyaml/kio"
 )
 
 type EvalOptions struct {
+	// FnResultList stores the result of the function evaluation
+	FnResultList fnresult.ResultList
```
Comment (martinmaly) [inline pkg/fn/eval.go:29]: this would need to be pointer (or opts below need to be pointer)

File: pkg/fn/eval.go [hunk 3]
```diff
@@ -17,13 +17,19 @@ package fn
...
 }
 
 type Evaluator interface {
 	Eval(ctx context.Context, pkg filesys.FileSystem, fn v1.Function, opts EvalOptions) error
+	NewRunner(ctx context.Context, fn *v1.Function, pkgPath types.UniquePath, opts EvalOptions) (kio.Filter, error)
```
Comment (martinmaly) [inline pkg/fn/eval.go:34]: what about something like:

```
type Evaluator /* maybe we should rename this */ interface {
  NewProcessor(ctx context.Context, fn v1.Function, opts) (ResourceListProcessor, error)
}
```

https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/fn/framework/framework.go#L67

so we don't need to split up the pieces of data (ResultList in opts, RNodes separately too)

File: pkg/fn/eval.go (hunk 2 above)
Comment (phanimarupaka) [reply pkg/fn/eval.go:28]: updated.

File: pkg/fn/eval.go (hunk 1 above)
Comment (phanimarupaka) [reply pkg/fn/eval.go:32]: pkgPath can be used for printing status messages. Removing it for now, you can reintroduce it based on the implementation.

File: pkg/fn/eval.go [hunk 4]
```diff
@@ -17,13 +17,18 @@ package fn
...
 )
 
 type EvalOptions struct {
+	// FnResultList stores the result of the function evaluation
+	FnResultList *fnresult.ResultList
 }
 
 type Evaluator interface {
 	Eval(ctx context.Context, pkg filesys.FileSystem, fn v1.Function, opts EvalOptions) error
+	NewRunner(ctx context.Context, fn *v1.Function, opts EvalOptions) (kio.Filter, error)
```
Comment (martinmaly) [inline pkg/fn/eval.go:33]: Couple suggestions:

NewRunner --> NewFilter
EvalOptions --> FilterOptions

File: pkg/fn/eval.go [hunk 5]
```diff
@@ -17,13 +17,18 @@ package fn
...
 
+	fnresult "github.com/GoogleContainerTools/kpt/pkg/api/fnresult/v1"
 	v1 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1"
 	"sigs.k8s.io/kustomize/kyaml/filesys"
+	"sigs.k8s.io/kustomize/kyaml/kio"
 )
 
 type EvalOptions struct {
+	// FnResultList stores the result of the function evaluation
+	FnResultList *fnresult.ResultList
```
Comment (martinmaly) [inline pkg/fn/eval.go:28]: I'd propose FnResultList --> ResultList

**Review by martinmaly — CHANGES_REQUESTED** [review]
couple nitpicks

File: pkg/fn/eval.go (hunk 4 above)
Comment (phanimarupaka) [reply pkg/fn/eval.go:33]: FilterOptions seems generic. NewRunner is consistent with fnruntime.

File: pkg/fn/eval.go (hunk 5 above)
Comment (phanimarupaka) [reply pkg/fn/eval.go]: updated.

File: pkg/fn/eval.go (hunk 4 above)
Comment (martinmaly) [reply pkg/fn/eval.go:33]: If we are going to stick with the fnruntime nomenclature, let's switch over completely:

Evaluator --> FunctionRunner
eval.go --> runner.go
EvalOptions --> RunnerOptions

wdyt?

I still wish NewRunner returned Runner and not Filter, but oh well :)

**Review by martinmaly — APPROVED** [review]

**Pull request merged**